src/dbnode/storage/bootstrap/process.go

```diff
@@ -266,24 +272,23 @@ func (b bootstrapProcess) Run(
 				continue
 			}
 
-			// Check if snapshot-type ranges have advanced while bootstrapping previous ranges.
+			// If second run, check if snapshot-type ranges have advanced while bootstrapping previous ranges.
 			// If yes, return an error to force a retry
-			if persistConf := ns.DataRunOptions.RunOptions.PersistConfig(); persistConf.Enabled &&
-				persistConf.FileSetType == persist.FileSetSnapshotType {
+			if runIndex == 1 {
```

Comment:
I think checking for the second run should be enough here to replace previous condition. This is needed because now second run blocks won't always have `persist.FileSetSnapshotType` set.

Reply:
Do I understand correctly that this is not really required to avoid loading blocks in memory for read only? Can we split this in to commits? Mainly because with my limited knowledge this seem to be a dpangerious change that might also affect non read only namespaces.

Reply:
This is needed because if this condition is left unchanged, we won't be able to check if target ranges have advanced for read-only namespaces. Don't think that this change is dangerous, because it will behave in the same way as before (just the condition is different)

Reply:
Hm, i agree the logic is still similar but i agree - i’d rather not change setting file snapshot type to something other than Snapshot.

I’ll keep reviewing and give recommendation once understanding the larger change, but seems we are going from type safe to type unsafe potentially.

Reply:
Would this make more sense if we based the decision on `"snapshotEnabled": false` namespace property (instead of `readOnly`)?

Reply:
Agree that decision based on `"snapshotEnabled": false` could be considered as well. 
And for Rob's concerns - I don't think type safety gives us much here. The most important thing we're doing here is checking that target ranges have not advanced for the *last* run or latest ranges we've calculated before. I am not sure if this should depend on snapshot fileSet type because this is more of the time problem we're solving here and not the type of persistence.

Reply:
Updated PR to use `snapshotEnabled` instead.